app/src/main/java/de/hhn/aib/labsw/blackmirror/controller/PageController.java

```diff
@@ -11,16 +16,19 @@
  *
  * @author Niklas Binder
  * @author Markus Marewitz
+ * @author Philipp Herda
  * @version 2022-05-11
  */
-public class PageController {
+public class PageController extends AbstractWidgetController {
```

Comment:
the page controller is not a controller for a specific widget and should not extend their abstract controller. If there is common functionality between this class and the abstract widget controller than you should factor out this logic from the abstract widget controller into another abstract parent class (maybe "AbstractAppInterface") to both the widget controller and this class, like so:
![app-interface-on-mirror-schema](https://user-images.githubusercontent.com/66299529/172880959-31c34ee3-d286-4e3c-b7b2-0ac47319ab55.jpg)

Reply:
I choose to pull the relevant method (nodeToObject) into the TopicListener instead and let PageController implement TopicListener and work with the MirrorApi Instance instead. Putting an abstract interface class thingy as an inbetween layer would be somewhat tricky since TopicListener is already an Interface :D
I was kinda aware that the original solution wasn´t really intended...